app/controllers/AccountManagement.scala

```diff
@@ -177,49 +179,65 @@ object ManageWeekly extends ContextLogging {
-            sequence(renewalPlans.map { plan =>
-              val price = plan.charges.price.getPrice(existingCurrency).toRight(s"could not find price in $existingCurrency for plan ${plan.id} ${plan.name}").right
-              price.map(price => WeeklyPlanInfo(plan.id, plan.charges.prettyPricing(price.currency)))
-            }).right.map(r => (existingCurrency, r))
-          }
-          weeklyPlanInfo match {
-            case Left(errorMessage) =>
-              error(s"couldn't get new rate/currency for renewal: $errorMessage")
-              Ok(views.html.account.details(None, promoCode, Some("We found your subscription, but can't renew it via the web, please contact customer services for help.")))
-            case Right((existingCurrency, weeklyPlanInfoList)) =>
-              Ok(weeklySubscription.asRenewable.map { renewableSub =>
-                info(s"sub is renewable - showing weeklyRenew page")
-                views.html.account.weeklyRenew(renewableSub, account.soldToContact, account.billToContact.email, billToCountry, weeklyPlanInfoList, existingCurrency, promoCode)
-              } getOrElse {
-                info(s"sub is not renewable - showing weeklyDetails page")
-                views.html.account.weeklyDetails(weeklySubscription, billingSchedule, account.soldToContact)
-              })
-          }
-        }.getOrElse {
-          error(s"no valid bill to country for account")
-          Ok(views.html.account.details(None, promoCode, Some("We found your subscription, but can't manage it via the web, please contact customer services for help.")))
-        }
-      }.run.map(_.fold({ errorMessage =>
-        error(s"problem getting account: $errorMessage")
-        Ok(views.html.account.details(None, promoCode, Some("We found your subscription, but can't manage it via the web, please contact customer services for help.")))
-      }, identity))
-    } else {
+
+    def getRenewalPlans(account: ZuoraRestService.AccountSummary, currency: Currency) = {
+      val catalog = tpBackend.catalogService.unsafeCatalog
+      // Identifies customers who have been wrongly migrated in on a price-adjusted Zone B or C rate plan and are
+      // having the paper delivered to the UK or USA, and pay by GBP or USD respectively. These customers should
+      // be offered Zone A rate plans upon renewal.
+      def switchToZoneAIfNecessary = account.soldToContact.country.find { soldToCountry =>
+        (soldToCountry == UK && currency == GBP) || (soldToCountry == US && currency == USD)
+      }.map(_ => catalog.weekly.zoneA.plans)
+
+      // Determine the rate plans to offer for renewal based on what product the customer has currently
+      def usePlansBasedOnExistingProduct = weeklySubscription.planToManage.product match {
+        case Product.WeeklyZoneA => catalog.weekly.zoneA.plans
+        case Product.WeeklyZoneB => catalog.weekly.zoneB.plans
+        case Product.WeeklyZoneC => catalog.weekly.zoneC.plans
+      }
```

Comment:
I feel like it would be clearer to keep the `catalog.weekly.zoneX.plans` decision in one place.
would it be better to put the 'shouldBeInZoneA' as a boolean, then pattern match on both in the same match statement?
so that would be change
```scala
      def switchToZoneAIfNecessary = account.soldToContact.country.find { soldToCountry =>
        (soldToCountry == UK && currency == GBP) || (soldToCountry == US && currency == USD)
      }.map(_ => catalog.weekly.zoneA.plans)

      // Determine the rate plans to offer for renewal based on what product the customer has currently
      def usePlansBasedOnExistingProduct = weeklySubscription.planToManage.product match {
        case Product.WeeklyZoneA => catalog.weekly.zoneA.plans
        case Product.WeeklyZoneB => catalog.weekly.zoneB.plans
        case Product.WeeklyZoneC => catalog.weekly.zoneC.plans
      }

      val weeklyPlans = switchToZoneAIfNecessary getOrElse usePlansBasedOnExistingProduct
```
to
```scala
      def shouldBeInZoneA = account.soldToContact.country.exists { soldToCountry =>
        (soldToCountry == UK && currency == GBP) || (soldToCountry == US && currency == USD)
      }

      val weeklyPlans = (weeklySubscription.planToManage.product, shouldBeZoneA) match {
        case (_, true) => catalog.weekly.zoneA.plans
        case (Product.WeeklyZoneA, _) => catalog.weekly.zoneA.plans
        case (Product.WeeklyZoneB, _) => catalog.weekly.zoneB.plans
        case (Product.WeeklyZoneC, _) => catalog.weekly.zoneC.plans
      }
```